db/version_set.cc

```diff
@@ -3647,6 +3668,11 @@ Status VersionSet::DumpManifest(Options& options, std::string& dscname,
         cfd->SetLogNumber(edit.log_number_);
       }
 
+      if (cfd != nullptr && edit.has_min_log_number_to_keep_ &&
+          edit.min_log_number_to_keep_ > latest_min_log_number_to_keep_) {
+        latest_min_log_number_to_keep_ = edit.min_log_number_to_keep_;
```

Comment:
A few lines below we call MarkMinLogNumberToKeep which does the same thing. The only difference is that over there it is not protected by (cfd != nullptr). What is the reason for the duplication?